pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -131,30 +149,133 @@ var _ = Describe("MachineSet Reconciler", func() {
+					}
+
+					return g.Expect(localInstance.Status.Conditions).Should(ContainElement(SatisfyAll(
+						HaveField("Type", Equal(PausedCondition)),
+						HaveField("Status", Equal(corev1.ConditionTrue)),
+					)))
+
+				},
+				// Condition / until function: until we observe the MachineAuthority being MAPI
+				func(_ context.Context, g framework.GomegaAssertions) bool {
+					By("Checking that the AuthoritativeAPI is not MachineAPI")
+
+					localInstance := instanceCopy.DeepCopy()
+					if err := k8sClient.Get(ctx, objectKey(localInstance), localInstance); err != nil {
+						return g.Expect(err).Should(WithTransform(isRetryableAPIError, BeTrue()), "expected temporary error while getting machine: %v", err)
+					}
+
+					return g.Expect(localInstance.Status.AuthoritativeAPI).ToNot(Equal(machinev1.MachineAuthorityMachineAPI))
+				})
+		}()
+
+		By("Updating the AuthoritativeAPI from Migrating to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMachineAPI
+		})).Should(Succeed())
+
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status.AuthoritativeAPI", Equal(machinev1.MachineAuthorityMachineAPI)))
+
+		By("Verifying the paused condition is approproately set to false")
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status.Conditions", ContainElement(SatisfyAll(
+			HaveField("Type", Equal(PausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionFalse)),
+		))))
+
+		By("Unsetting the AuthoritativeAPI field in the status")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = ""
+		})).Should(Succeed())
+
+		By("Verifying the paused condition is still approproately set to false")
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status.Conditions", ContainElement(SatisfyAll(
+			HaveField("Type", Equal(PausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionFalse)),
+		))))
+	})
 })
 
-func cleanResources() error {
-	machineSets := &machinev1.MachineSetList{}
-	if err := k8sClient.List(ctx, machineSets); err != nil {
+func cleanResources(namespace string) error {
+	machineSet := &machinev1.MachineSet{}
+	if err := k8sClient.DeleteAllOf(ctx, machineSet, client.InNamespace(namespace)); err != nil {
 		return err
 	}
-	for _, machineSet := range machineSets.Items {
-		ms := machineSet
-		if err := k8sClient.Delete(ctx, &ms); err != nil {
-			return err
-		}
-	}
 
-	machines := &machinev1.MachineList{}
-	if err := k8sClient.List(ctx, machines); err != nil {
+	machine := &machinev1.Machine{}
+	if err := k8sClient.DeleteAllOf(ctx, machine, client.InNamespace(namespace)); err != nil {
 		return err
 	}
-	for _, machine := range machines.Items {
-		m := machine
-		if err := k8sClient.Delete(ctx, &m); err != nil {
-			return err
-		}
-	}
 
 	return nil
 }
+
+func objectKey(object metav1.Object) client.ObjectKey {
```

Comment:
I think the client package has a function that does this already that you could use